service/src/LevelZero.hpp

```diff
@@ -249,8 +249,17 @@ namespace geopm
                                                     int l0_domain,
                                                     int l0_domain_idx,
                                                     double setting) const = 0;
+
+            //Level Zero Tools/ZET
+            virtual std::vector<double> metric_sample(unsigned int l0_device_idx,
+                                                      unsigned int l0_domain_idx,
+                                                      std::string metric_name) const = 0;
+            virtual uint32_t metric_update_rate(unsigned int l0_device_idx) const = 0;
+
+            virtual void metric_read(unsigned int l0_device_idx,
+                                     unsigned int l0_domain_idx) = 0;
     };
 
-    const LevelZero &levelzero();
+    LevelZero &levelzero();
```

Comment:
This change was required for the lazy init approach being used to update some level zero metric containers.  After more recent code changes it might not be required.